samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -0,0 +1,192 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# The Gadgets Profile relies on the following external dependencies that
+# must be downloaded and run before the build stage:
+# - nanopb generator
+# - protocol buffer definitions
+
+# Note: as there doesn't seem to be an easy way of checking the version of the nanopb generator (protoc-gen-nanopb binary)
+# there is the possibility of a version mismatch between the generator and the nanopb source files that is used in the firmware sample.
+# If there is a mismatch, there will be a compilation error in the build step.
+# The easiest way to fix such an error is to download the appropriate nanopb generator .zip and extract it to this sample folder.
+# (https://jpa.kapsi.fi/nanopb/download/)
+
+# Check local folder for protoc and nanopb binaries presence
+file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc")
+if (NOT LOCAL_PROTOC_BIN)
+    file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc.exe")
+endif()
+file(GLOB_RECURSE LOCAL_NANOPB_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc-gen-nanopb*")
+
+# Check system for protoc and nanopb binaries presence
+find_program(PROTOC_BIN protoc)
+find_program(NANOPB_BIN protoc-gen-nanopb)
+
+# Prefer local binaries to make version matching with nanopb source repo easier
+if (LOCAL_PROTOC_BIN)
+    set(PROTOC_BIN ${LOCAL_PROTOC_BIN})
+endif()
+
+if (LOCAL_NANOPB_BIN)
+    set(NANOPB_BIN ${LOCAL_NANOPB_BIN})
+endif()
+
+if ((NOT NANOPB_BIN) OR (NOT PROTOC_BIN))
+    message(STATUS "nanopb not found.")
+    message(STATUS "Please download nanopb from here https://jpa.kapsi.fi/nanopb/download/")
+    message(STATUS "and extract to either this sample directory or somewhere cmake can find it via system PATH.")
+    message(FATAL_ERROR "nanopb required by this project.")
+endif()
+
+message(STATUS "Using protoc: ${PROTOC_BIN}")
+message(STATUS "Using protoc-gen-nanopb: ${NANOPB_BIN}")
+
+if(ALEXA_EMBEDDED_PATH)
+  # Do nothing, just use the provided path
+elseif(WEST)
+  ## Use `west list` to find the alexa-embedded path
+  execute_process(
+    COMMAND
+    ${WEST} list Alexa-Gadgets-Embedded-Sample-Code --format={posixpath}
+    OUTPUT_VARIABLE
+    west_project_output
+    RESULT_VARIABLE
+    result
+  )
+  string(REGEX REPLACE "[\r\n]+" "" ALEXA_EMBEDDED_PATH "${west_project_output}")
+  if(${result})
+    message(FATAL_ERROR "Failed to find alexa-embedded, cannot build sample")
+  endif()
+else()
+  message(FATAL_ERROR "west not installed, please provide ALEXA_EMBEDDED_PATH to CMake to build this sample")
+endif()
+
+if(NANOPB_PATH)
+  # Do nothing, just use the provided path
+elseif(WEST)
+  ## Use `west list` to find the nanopb path
+  execute_process(
+    COMMAND
+    ${WEST} list nanopb --format={posixpath}
```

Comment:
as per:  https://github.com/zephyrproject-rtos/zephyr/pull/26060
`west` should not be called directly.